Gems/Atom/RPI/Code/Source/RPI.Public/Buffer/Buffer.cpp

```diff
@@ -136,16 +136,24 @@ namespace AZ
 
             if (resultCode == RHI::ResultCode::Success)
             {
-                m_bufferAsset = { &bufferAsset, AZ::Data::AssetLoadBehavior::PreLoad };
                 InitBufferView();
                 
                 if (bufferAsset.GetBuffer().size() > 0 && !initWithData)
                 {
+                    m_bufferAsset = { &bufferAsset, AZ::Data::AssetLoadBehavior::PreLoad };
+
                     AZ_PROFILE_SCOPE(RPI, "Stream Upload");
                     m_streamFence = RHI::Factory::Get().CreateFence();
                     if (m_streamFence)
                     {
                         m_streamFence->Init(m_rhiBufferPool->GetDevice(), RHI::FenceState::Reset);
+                        m_streamFence->WaitOnCpuAsync(
+                            [this]()
+                            {
+                                this->m_pendingUploadMutex.lock();
+                                this->m_bufferAsset.Reset();
+                                this->m_pendingUploadMutex.unlock();
+                            });
```

Comment:
Can you add a comment about why the buffer asset needs to be reset in async manner. Thanks.